core/include/kernel/interrupt.h

```diff
@@ -132,8 +138,7 @@ static inline bool itr_chip_is_valid(struct itr_chip *chip)
 	       is_unpaged((void *)chip->ops) &&
 	       chip->ops->mask && is_unpaged(chip->ops->mask) &&
 	       chip->ops->unmask && is_unpaged(chip->ops->unmask) &&
-	       chip->ops->enable && chip->ops->disable &&
-	       chip->ops->add;
```

Comment:
I'd rather have a separate version of `itr_chip_is_valid()` for the DT use case, for clarity and consistent checking.